src/integrationtests/java/com/aws/greengrass/integrationtests/SyncTest.java

```diff
@@ -191,6 +192,40 @@ void GIVEN_sync_config_and_no_local_WHEN_startup_THEN_local_version_updated_via_
                 any(software.amazon.awssdk.services.iotdataplane.model.UpdateThingShadowRequest.class));
     }
 
+    @ParameterizedTest
+    @ValueSource(classes = {RealTimeSyncStrategy.class, PeriodicSyncStrategy.class})
+    void GIVEN_sync_config_WHEN_repeat_turn_sync_on_and_off_THEN_no_exceptions_thrown(Class<?
+            extends BaseSyncStrategy> clazz, ExtensionContext context) throws InterruptedException {
+        Level l = LogConfig.getRootLogConfig().getLevel();
+        LogConfig.getRootLogConfig().setLevel(Level.ERROR); // set to ERROR level to avoid spamming logs
+        try {
+            GetThingShadowResponse shadowResponse = mock(GetThingShadowResponse.class, Answers.RETURNS_DEEP_STUBS);
+            lenient().when(shadowResponse.payload().asByteArray()).thenReturn(cloudShadowContentV10.getBytes(UTF_8));
+
+            // existing document
+            when(iotDataPlaneClientFactory.getIotDataPlaneClient().getThingShadow(any(GetThingShadowRequest.class))).thenReturn(shadowResponse);
+
+            startNucleusWithConfig(
+                    NucleusLaunchUtilsConfig.builder().configFile(getSyncConfigFile(clazz)).mockCloud(true).syncClazz(clazz).build());
+
+            ShadowManager.StartSyncInfo syncInfo =
+                    ShadowManager.StartSyncInfo.builder().reInitializeSyncInfo(false).overrideRunningCheck(true).updateCloudSubscriptions(false).build();
+
+            SyncHandler syncHandler = kernel.getContext().get(SyncHandler.class);
+            Instant end = Instant.now().plusSeconds(30);
+            // this is more of a sanity check to ensure that cancelling threads doesn't cause any errors
+
+            // stop and start syncing for 30 seconds. Sync requests will be added when syncing starts.
+            while (end.isAfter(Instant.now())) {
```

Comment:
should we have shadow operations on a separate thread happening here?

Reply:
they are happening on a separate thread - when syncing starts is uses the ExecutorService to start syncing on another thread

Reply:
it's got some full sync operations that occur already

Reply:
Ok right. I forgot that even the Full sync request would get the sync info and shadow on even if there is nothing to really update. So that should touch the DB.

Reply:
I thought about it some more and added a thread that does random updates, as well as sleeping for random times to allow syncing to occur for various lengths of time